cocoa/src/toga_cocoa/widgets/optioncontainer.py

```diff
@@ -1,98 +1,98 @@
+    def set_bounds(self, x, y, width, height):
+        super().set_bounds(x, y, width, height)
 
-        Args:
-            index: The index in the tab list where the tab should be added.
-            text (str): The text for the option container
-            widget: The widget or widget tree that belongs to the text.
-        """
-        container = Container(content=widget)
+        # Setting the bounds changes the constraints, but that doesn't mean
+        # the constraints have been fully applied. Schedule a refresh to be done
+        # as soon as possible in the future
+        self.native.performSelector(
+            SEL("refreshContent"), withObject=None, afterDelay=0
+        )
 
+    def add_content(self, index, text, widget):
+        # Establish the minimum layout
+        widget.interface.style.layout(widget.interface, MinimumContainer())
+        min_width = widget.interface.layout.width
+        min_height = widget.interface.layout.height
+
+        # Create the container for the widget
+        container = Container()
+        container.content = widget
+        container.min_width = min_width
+        container.min_height = min_height
+        self.sub_containers.insert(index, container)
+
+        # Create a NSTabViewItem for the content
         item = NSTabViewItem.alloc().init()
         item.label = text
 
-        # Turn the autoresizing mask on the widget widget
-        # into constraints. This makes the widget fill the
-        # available space inside the OptionContainer.
-        container.native.translatesAutoresizingMaskIntoConstraints = True
-
         item.view = container.native
         self.native.insertTabViewItem(item, atIndex=index)
 
     def remove_content(self, index):
         tabview = self.native.tabViewItemAtIndex(index)
-        if tabview == self.native.selectedTabViewItem:
-            # Don't allow removal of a selected tab
-            raise self.interface.OptionException(
-                "Currently selected option cannot be removed"
-            )
-
         self.native.removeTabViewItem(tabview)
 
-    def set_on_select(self, handler):
-        pass
+        sub_container = self.sub_containers[index]
+        sub_container.content = None
+        del self.sub_containers[index]
 
     def set_option_enabled(self, index, enabled):
         tabview = self.native.tabViewItemAtIndex(index)
         if enabled:
             try:
                 self._disabled_tabs.remove(index)
             except KeyError:
+                # Enabling a tab that wasn't previously disabled
                 pass
         else:
-            if tabview == self.native.selectedTabViewItem:
-                # Don't allow disable a selected tab
-                raise self.interface.OptionException(
-                    "Currently selected option cannot be disabled"
-                )
-
             self._disabled_tabs.add(index)
         tabview._setTabEnabled(enabled)
```

Comment:
#832 added the enable/disable feature using some undocumented APIs `_isTabEnabled` and `_setTabEnabled`. #1117 then removed the use of the `is` method, but left the `set` method alone, and added a comment saying "it's handled by the delegate" even though it actually isn't. This is very confusing.

Reply:
My feeling is that if there are some undocumented methods which have worked all this time, then we might as well keep on using them, but wrap them with broad exception handlers in case the methods ever change or disappear in a future version of macOS. That way, we'll still discover such problems as soon as we start running the testbed on the new macOS version, but existing Toga-based apps will only display a warning rather than crash.

Reply:
Yeah - this is weird. I'm not sure how anyone found the original `_setTabEnabled` method. I've added 2 layers of protection: catching the AttributeError in case the method disappears; and implementing the (missing, but referenced) delegate method that is strictly responsible for determining whether the tab can be selected.